core/runtime/src/main/java/io/quarkiverse/langchain4j/guardrails/OutputGuardrail.java

```diff
@@ -122,7 +50,8 @@ default void validate(OutputGuardrailParams params)
      * @param memory the memory, can be {@code null} or empty
      * @param augmentationResult the augmentation result, can be {@code null}
      */
-    record OutputGuardrailParams(AiMessage responseFromLLM, ChatMemory memory, AugmentationResult augmentationResult) {
+    record OutputGuardrailParams(AiMessage responseFromLLM, ChatMemory memory,
+            AugmentationResult augmentationResult) implements GuardrailParams {
```

Comment:
I didn't know that records could implement interfaces. That may explain why the interface is just a marker. Can we try to define memory() and augmentationResult() in the interface? Would that work? 

(again discovering this feature)

Reply:
Done.

Reply:
I learn a thing about records today!